functions/src/services/symptomScore/defaultSymptomScoreCalculator.test.ts

```diff
@@ -6,25 +6,111 @@
 // SPDX-License-Identifier: MIT
 //
 
+import fs from 'fs'
 import { expect } from 'chai'
 import { describe } from 'mocha'
 import { DefaultSymptomScoreCalculator } from './defaultSymptomScoreCalculator.js'
-import { mockQuestionnaireResponse } from '../../tests/mocks/questionnaireResponse.js'
+import { average } from '../../extensions/array.js'
+import { FHIRQuestionnaireResponse } from '../../models/fhir/fhirQuestionnaireResponse.js'
 
 describe('DefaultSymptomScoreCalculator', () => {
-  it('should calculate symptom score', () => {
-    const response = mockQuestionnaireResponse()
+  it('correctly computes symptom scores as described in the resource file', () => {
     const calculator = new DefaultSymptomScoreCalculator()
-    const score = calculator.calculate(response)
-    expect(score).to.deep.equal({
-      questionnaireResponseId: response.id,
-      date: response.authored,
-      overallScore: 54,
-      socialLimitsScore: 58,
-      physicalLimitsScore: 42,
-      qualityOfLifeScore: 63,
-      symptomFrequencyScore: 52,
-      dizzinessScore: 3,
-    })
+    const fileContent = fs.readFileSync(
+      'src/tests/resources/symptomScores.csv',
+      'utf8',
+    )
+    const lines = fileContent.split('\n').slice(1)
+    expect(lines.length).to.equal(2267)
+
+    let lineIndex = -1
+    for (const line of lines) {
+      lineIndex++
+      const values = line.split(',')
+      expect(values.length).to.equal(18)
+
+      const answer9 = (lineIndex % 6) - 1
+      const response = FHIRQuestionnaireResponse.create({
+        questionnaire:
+          'http://spezi.health/fhir/questionnaire/9528ccc2-d1be-4c4c-9c3c-19f78e51ec19',
+        questionnaireResponse: '60193B69-B0F7-4708-8CCE-F3CB2936628D',
+        date: new Date('2024-07-18T18:46:37.219581961-07:00'),
+        answer1a: parseInt(values[0]),
+        answer1b: parseInt(values[1]),
+        answer1c: parseInt(values[2]),
+        answer2: parseInt(values[3]),
+        answer3: parseInt(values[4]),
+        answer4: parseInt(values[5]),
+        answer5: parseInt(values[6]),
+        answer6: parseInt(values[7]),
+        answer7: parseInt(values[8]),
+        answer8a: parseInt(values[9]),
+        answer8b: parseInt(values[10]),
+        answer8c: parseInt(values[11]),
+        answer9: answer9,
+      })
+
+      const score = calculator.calculate(response)
+      expect(score.questionnaireResponseId).to.equal(response.id)
+      expect(score.date).to.equal(response.authored)
+
+      if (score.physicalLimitsScore !== undefined) {
+        expect(
+          score.physicalLimitsScore,
+          `${lineIndex} - Physical Limits: ${line}`,
+        ).to.approximately(parseFloat(values[12]), 0.16667)
```

Comment:
Would be good to add in-line comments & reasoning behind the proximity estimations for now so people can figure out why this was the case later on. Applies to others as well.